### PR TITLE
sile: build and install documentation (self test)

### PR DESCRIPTION
**Pull request by ck3d**
###### Motivation for this change
Documentation and tests for Sile was missing.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by markuskowa** [comment]
@GrahamcOfBorg eval

**Review by markuskowa — COMMENTED** [review]
The addition of the fonts could be mentioned in the commit message.

**Comment by ck3d** [comment]
Thanks. I added a note to explain the addition of fonts.

File: pkgs/tools/typesetting/sile/default.nix [hunk 1]
```diff
@@ -40,15 +41,31 @@ stdenv.mkDerivation rec {
...
+  FONTCONFIG_FILE = makeFontsConf {
+    fontDirectories = [
+      gentium
+      gentium-book-basic
+      dejavu_fonts
+    ];
+  };
+
+  buildPhase = ''
+   make all documentation/developers.pdf documentation/sile.pdf
```
Comment (markuskowa) [inline pkgs/tools/typesetting/sile/default.nix]: Sorry I forgot that comment: What about putting it in `postBuild`? That would leave the nix-magic of the buildPhase intact.
```
postBuild = ''
  make documentation/developers.pdf documentation/sile.pdf
'';
```

**Comment by markuskowa** [comment]
@GrahamcOfBorg build sile

**Comment by markuskowa** [comment]
@GrahamcOfBorg build sile

**Pull request merged**